### PR TITLE
Dawn Spider

### PR DESCRIPTION
**Pull request by zain14shah**

### CODE REVIEW

**Review by hamza-jutt — CHANGES_REQUESTED** [review]
Data extracted should not contain tags unless required (That usually never happens). Why is there commented lines in final revision? Fix the issues.

**Comment by zain14shah** [comment]
Having an issue applying enumerate().
enumerate

File: prayers_timings.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        'https://www.dawn.com/prayers-timings'
+    ]
+
+    def parse(self, response):
+        prayer_names = response.xpath(
+            '//table[@class ="table  table-bordered table-striped"]//th/text()').extract()[1:]
+
+        for prayer_data in response.xpath('//tr[@class ="text-center"]'):
+            city_name = prayer_data.xpath(
+                './/td/strong/text()').extract_first()
```
Comment (hamza-jutt) [inline prayers_timings.py]: can be on same line

File: prayers_timings.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+class PrayersTimingsSpider(scrapy.Spider):
+    name = 'prayers_timings'
+
+    start_urls = [
+        'https://www.dawn.com/prayers-timings'
+    ]
+
+    def parse(self, response):
+        prayer_names = response.xpath(
+            '//table[@class ="table  table-bordered table-striped"]//th/text()').extract()[1:]
```
Comment (hamza-jutt) [inline prayers_timings.py]: A better formatting that looks clean can be 
```
function_name(
    'lengthy_argument_1', 'lengthy_argument_2'
)
```

File: prayers_timings.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+        for prayer_data in response.xpath('//tr[@class ="text-center"]'):
+            city_name = prayer_data.xpath(
+                './/td/strong/text()').extract_first()
+            all_prayer_timings = prayer_data.xpath('.//td/text()').extract()
+
+            index = 0
```
Comment (hamza-jutt) [inline prayers_timings.py]: use enumerate to maintain indexes.

File: prayers_timings.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+            prayers = {}
+            for name in prayer_names:
+                prayers.update({name: all_prayer_timings[index]})
```
Comment (hamza-jutt) [inline prayers_timings.py]: can use `prayers[name] = all_prayer_timings[index]`

File: prayers_timings.py [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+                index = index + 1
+            yield ({city_name: prayers})
```
Comment (hamza-jutt) [inline prayers_timings.py]: can be `yield {city_name: prayers}`

File: prayers_timings.py [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+        prayer_names = response.xpath(
+            '//table[@class ="table  table-bordered table-striped"]//th/text()').extract()[1:]
+
+        for prayer_data in response.xpath('//tr[@class ="text-center"]'):
+            city_name = prayer_data.xpath(
+                './/td/strong/text()').extract_first()
+            all_prayer_timings = prayer_data.xpath('.//td/text()').extract()
+
+            index = 0
+            prayers = {}
```
Comment (hamza-jutt) [inline prayers_timings.py]: shoud be `prayer_timings = {}`

**Review by hamza-jutt — CHANGES_REQUESTED** [review]
Read the docs for enumerate issue that you are facing and update the code.

**Review by hamza-jutt — APPROVED** [review]

**Pull request merged**